### PR TITLE
Update default splitting behavior to get fewer, larger, splits

### PR DESCRIPTION
**Pull request by kennknowles**
Before this change, the portable Java splitting of BoundedSource was capped at 64k. Changed to match the Python behavior, which has some fairly reasonable default split sizes that seem to work well in practice.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by kennknowles** [comment]
CC @bhisevishal

**Comment by kennknowles** [comment]
R: @robertwb

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/Read.java [hunk 1]
```diff
@@ -290,12 +289,17 @@ public void splitRestriction(
       long estimatedSize = restriction.getEstimatedSizeBytes(pipelineOptions);
       // Split into pieces as close to the default desired bundle size but if that would cause too
       // few splits then prefer to split up to the default desired number of splits.
-      long splitBundleSize =
-          Math.min(
-              DEFAULT_DESIRED_BUNDLE_SIZE_BYTES,
-              Math.max(1L, estimatedSize / DEFAULT_DESIRED_NUM_SPLITS));
```
Comment (robertwb) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/Read.java:296]: Should we delete the DEFAULT_DESIRED_NUM_SPLITS constant as well?

**Review by robertwb — APPROVED** [review]

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/Read.java (hunk 1 above)
Comment (kennknowles) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/io/Read.java:296]: It is used in the unbounded source wrapper, but I have little context on that.

**Comment by kennknowles** [comment]
Run Java PreCommit

**Comment by kennknowles** [comment]
Run Java PreCommit

**Comment by kennknowles** [comment]
Run Java_GCP_IO_Direct PreCommit

**Comment by kennknowles** [comment]
GCP IO is perma-red due to firestore - looks like it went red without code changes so probably our project got messed up

**Comment by kennknowles** [comment]
Hmm this appears to be a common flake.

**Pull request merged**